python/test/unit/language/test_tlx.py

```diff
@@ -2217,3 +2217,89 @@ def asycn_copy_kernel(x_ptr, y_ptr, cond):
     kernel = asycn_copy_kernel[grid](x, y, True)
     assert kernel.asm["ttgir"].count("ttg.async_copy_global_to_local") == 2
     assert kernel.asm["ttgir"].count("ttg.async_commit_group") == 1
+
+
+@pytest.mark.parametrize(
+    "src_dtype, dst_dtype",
+    [
+        ("float32", "float8_e5m2"),
+        ("float32", "float8_e4m3fn"),
+        ("float32", "float16"),
+        ("float32", "bfloat16"),
+    ],
+)
+def test_stoch_round(src_dtype, dst_dtype, device):
+
+    @triton.jit
+    def stoch_round_kernel(x_ptr, y_ptr, BLOCK_SIZE: tl.constexpr):
+        offsets = tl.arange(0, BLOCK_SIZE)
+        x = tl.load(x_ptr + offsets)
+        # Generate 1/4 shape for each random stream
+        offsets_quarter = tl.arange(0, BLOCK_SIZE // 4)
+        r0, r1, r2, r3 = tl.randint4x(0, offsets_quarter, n_rounds=7)
+        # Combine the 4 blocks into a single vector of random values
+        # r0,r1,r2,r3: each [BLOCK_SIZE//4]
+        # after joins: rbits: [BLOCK_SIZE]
+        rbits = tl.join(tl.join(r0, r1), tl.join(r2, r3)).reshape(x.shape)
+        y = tlx.stoch_round(
+            x,
+            tlx.dtype_of(y_ptr),
+            rbits,
+        )
+        tl.store(y_ptr + offsets, y)
+
+    # Map string names to torch dtypes
+    dtype_map = {
+        "float32": torch.float32,
+        "float16": torch.float16,
+        "bfloat16": torch.bfloat16,
+        "float8_e5m2": torch.float8_e5m2,
+        "float8_e4m3fn": torch.float8_e4m3fn,
+    }
+
+    src_dtype_torch = dtype_map[src_dtype]
+    dst_dtype_torch = dtype_map[dst_dtype]
+
+    SIZE = 256
+    a = torch.randn([SIZE], dtype=torch.float32, device=device).to(src_dtype_torch)
```

Comment:
Should we seed the random input to avoid flakey unit tests or do we purposely want this to test random inputs? Maybe we could include some debug information on how to reproduce because I worry if this fails it will be really tough to fix.